src/main/java/tatracker/logic/commands/session/EditSessionCommand.java

```diff
@@ -191,11 +194,11 @@ public void setEndTime(LocalDateTime endTime) {
         }
 
         public void setRecurring(int isRecurring) {
-            this.isRecurring = isRecurring;
+            this.newRecurring = isRecurring;
         }
 
-        public int getRecurring() {
-            return this.isRecurring;
+        public Optional<Integer> getRecurring() {
+            return Optional.ofNullable(newRecurring);
```

Comment:
Hmm no you would need to do:

if newRecurring < 0 return Optional.empty
else return Optional.of(newRecurring)